### DISCUSSION TITLE
How to modify the default Fedora image used in machine init under Windows?

### DISCUSSION THREAD

**Discussion by apollondor** [containers/podman#21787] [General]
Hey folks,
Small note: I don't have much experience with virtualization, fcos etc

I found into some podman's man on the web the statement that the default used Fedora OS is actually slightly modified for Podman on WSL. As it looks it doesn't use a FCOS image there. Henceforth I want to keep whatever image is used there and build on top of it. 
How would that be possible?

Another note, I see there are ignition files which I figure are some kind of configuration start up files, but If I am not wrong ignitions are for FCOS images, no? If they still can be used then man pages state that there are some default ones used for building SSH keys etc and if our custom ones are to be used the default ones are lost.

All in all what is the best way going forward without breaking any of the hard work podman team has put into the image, ignition files and be able to have the OS customized to my needs whenever I run the machine init step?

Thank you!

**Comment by baude** [comment]
Hello, yes the story is sort of confusing; however, the general theme is that whatever is run in the podman machine is meant to be an appliance like approach.  While it is true, we are opening things up more for customizations, we also then make an assumption that if you want to go this route, you are somewhat on your own.  That said, here are some informations that might help.

QEMU, HyperV, and the Apple Hypervisor implementations all use a custom Fedora CoreOS image.  WSL, however, has some technical hurdles that need to be solved before it can run FCOS due our reliance on ignition and systemd; so it runs a custom Fedora OS.  If you wanted to customize that, you could consider taking what is provided for WSL and altering that.  Exporting and then tarballing that up and feeding it to `podman machine init` as your image.  Again, we make assumptions that you are kind of on your own at that point because the support matrix for such a thing like this is not tenable.

Good luck!

**Reply by apollondor** [reply]
Hey @baude, 
thank you for the time and reply!

Yeap I eventually figured that one out by finding the Podman WSL Github image and did the same as you were mentioning. All worked well. But how about the default ignition files podman is using, would you know where I could find them? Was trying to look them myself, but maybe I'm just a bitsy blind as I couldn't find them. 

On another note, if you are a contributor/maintainer to this project, would you know is there a plan to change this WSL image in the future for an atomic like one (e.g. Silverblue etc)? 
The basic goal of what I was trying to achieve here is give my team an image with all basic tools we need on a daily basis and ensure they always get the proper tools versions e.g. Proper kubectl version etc
